pkg/util/release/virtual_strategy.go

```diff
@@ -0,0 +1,98 @@
+package release
+
+import (
+	"math"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+)
+
+func BuildVirtualStrategy(strategy *shipper.RolloutStrategy, surgeWeight int) (*shipper.RolloutVirtualStrategy, error) {
+	rolloutVirtualStrategy := make([]shipper.RolloutStrategyVirtualStep, len(strategy.Steps))
+
+	// create the first virtual transition: from previous release to first strategy step
+	step := shipper.RolloutStrategyStep{
+		Capacity: shipper.RolloutStrategyStepValue{
+			Incumbent: 100,
+			Contender: 0,
+		},
+		Traffic: shipper.RolloutStrategyStepValue{
+			Incumbent: 100,
+			Contender: 0,
+		},
+	}
+	nextStep := strategy.Steps[0]
+	rolloutStrategyVirtualSteps := buildRolloutStrategyVirtualSteps(step, nextStep, surgeWeight)
+	rolloutVirtualStrategy[0] = shipper.RolloutStrategyVirtualStep{VirtualSteps: rolloutStrategyVirtualSteps}
+
+	for i, step := range strategy.Steps {
+		if i != len(strategy.Steps)-1 {
```

Comment:
I wonder if this could be better expressed as a range on `strategy.Steps[:len(strategy.Steps)-1]` or similar instead.